crates/eth-rpc/tests/rpc.rs

```diff
@@ -1013,4 +1015,19 @@ mod tests {
         );
         assert_eq!(U256::from(transaction.block_number.unwrap()), U256::from(13));
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 3)]
```

Comment:
i recreated the flow that was failing when we try to use forge for deploy, here.

so it is meant to fail, but i was able to reproduce a smaller minimal case that i created an issue for, so i can iterate and do a different test